trollpy/tests.py

```diff
@@ -0,0 +1,357 @@
+    request = testing.DummyRequest()
+    request.method = method
+    request.dbsession = new_session
+    return request
+
+
+@pytest.fixture
+def dummy_post_request(new_session, method="POST"):
+    """Instantiate a fake HTTP Request, complete with a database session."""
+    request = testing.DummyRequest()
+    request.method = method
+    request.dbsession = new_session
+    return request
+
+
+@pytest.fixture
+def testapp():
+    """Create an instance of webtests TestApp for testing routes."""
+    from webtest import TestApp
+    from trollpy import main
+
+    app = main({}, **{'sqlalchemy.url': os.environ["DATABASE_URL"]})
+    testapp = TestApp(app)
+    session_factory = app.registry["dbsession_factory"]
+    engine = session_factory().bind
+    Base.metadata.drop_all(bind=engine)
+    Base.metadata.create_all(bind=engine)
+
+    return testapp
+
+
+@pytest.fixture
+def fill_the_db(testapp):
+    """Fill the database with some model instances."""
+    session_factory = testapp.app.registry["dbsession_factory"]
+    with transaction.manager:
+        dbsession = get_tm_session(session_factory, transaction.manager)
+        user = User(username='amos',
+                    password='password',
+                    first_name='Amos',
+                    last_name='Boldor',
+                    email='amosboldor@gmail.com',
+                    admin=True)
+        kill_score = KillScore(killscore_id='-8',
+                               statement='Prepare to die.')
+        tts = gTTS(text='You live because I allowed it.', lang='en')
+        tts.save('hello.mp3')
+        current_place = os.path.dirname(os.path.abspath(__file__))[:-7]
+        with open(current_place + 'hello.mp3', 'rb') as a:
+            data = a.read()
+        mp3 = Audio(id=1, binary_file=data)
+        dbsession.add(mp3)
+        dbsession.add(kill_score)
+        dbsession.add(user)
+
+# ======== FUNCTIONAL TESTS ===========
+
+
+def test_home_page_pops_up(testapp):
+    """Test that home page get sent correctly."""
+    response = testapp.get('/', status=200)
+    assert response.status_code == 200
+
+
+def test_login_leads_to_home(testapp, fill_the_db):
+    """Test that after logging in it sends you to the home route."""
+    response = testapp.post('/login',
+                            params={'username': 'amos',
+                                    'password': 'password'}).follow()
+    assert response.status_code == 200
+
+
+def test_login_show_form(testapp):
+    """Test that the login route shows form."""
+    response = testapp.get('/login', status=200)
+    assert response.status_code == 200
+    assert response.html.find('form')
+
+
+def test_logout(testapp):
+    """Test that logging out has 200 ok code."""
+    testapp.post('/login',
+                 params={'username': 'amos',
+                         'password': 'password'})
+    response = testapp.get('/logout', status=302).follow()
+    assert response.status_code == 200
```

Comment:
This isn't testing that logging out has a 200 code. It tests that when you get redirected, the page that you land on gives you a 200 code.